util/Yay0decompress.py

```diff
@@ -12,27 +12,30 @@ def decompress_yay0(in_bytes, byte_order="big"):
     link_table_idx = link_table_offset
     chunk_idx = chunk_offset
     other_idx = 16
-    
+
     mask_bit_counter = 0
     current_mask = 0
 
-    ret = b""
+    # preallocate result and index into it
+    idx = 0
+    ret = [0] * decompressed_size;
```

Comment:
Can we allocate a bytearray here rather than converting it later?

Reply:
```sh
Traceback (most recent call last):
  File "tools/n64splat/util/Yay0decompress.py", line 75, in <module>
    main(args)
  File "tools/n64splat/util/Yay0decompress.py", line 63, in main
    decompressed = decompress_yay0(raw_bytes, args.byte_order)
  File "tools/n64splat/util/Yay0decompress.py", line 31, in decompress_yay0
    ret[idx] = in_bytes[chunk_idx]
TypeError: 'bytes' object does not support item assignment
```
computer says no 👎

Reply:
Looks like [bytes is immutable whilst bytearray is mutable](https://docs.python.org/3/library/functions.html#func-bytearray)
![image](https://user-images.githubusercontent.com/9429556/99449240-a1c59800-2917-11eb-870c-3e75cfacbd94.png)

Reply:
update to the PR incoming...

Reply:
think I have 1 more small change...
```py
            ret[idx:idx + count] = ret[offset + count]
            idx += count + 1
```

just testing

Reply:
ignore that.